### DISCUSSION TITLE
payment from stripe is failing

### DISCUSSION THREAD

**Discussion by kpr12947** [juspay/hyperswitch#4379] [Q&A] [answered]
input -
import requests

url = "https://sandbox.hyperswitch.io/payments"

payload = {
  "amount": 6540,
  "currency": "USD",
  "confirm": True,
  "capture_method": "automatic",
  "capture_on": "2022-09-10T10:11:12Z",
  "amount_to_capture": 6540,
  "customer_id": "cus_KkCoKDaIg1inPqpHzZ5R",
  "name": "org2",
  "phone": "999999999",
  "phone_country_code": "+1",
  "description": "Its my first payment request",
  "authentication_type": "no_three_ds",
  "payment_method": "card",
  "payment_method_type": "credit",
  "payment_method_data": {
    "card": {
      "card_number": "4242424242424242",
      "card_exp_month": "10",
      "card_exp_year": "25",
      "card_holder_name": "Doe",
      "card_cvc": "123"
    }
  }
}
headers = {
    "api-key": "<api-key>",
    "Content-Type": "application/json"
}

response = requests.request("POST", url, json=payload, headers=headers)

output -

Sending credit card numbers directly to the Stripe API is generally unsafe. We suggest you use test tokens that map to the test card you are using, see https://stripe.com/docs/testing. To enable raw card data APIs in test mode, see https://support.stripe.com/questions/enabling-access-to-raw-card-data-apis.

payment transaction failed by showing the above message. how can i fix it?

**Comment by prajjwalkumar17** [comment, marked as answer]
Hey @rosemaryjogn 👋,
So previously we used to have an option in Stripe dashboard that allowed us to bypass the security rule of not passing the `raw-card details`. But in a very recent change they have depricated the feature so currently Stripe doesn't supports any payment made through raw card. You can try our flows with some other connectors possibly `Adyen`, `Noon`, `Trustpay` etc there are a lot of option to choose from.

**Reply by Kamal-Moha** [reply]
Hi @prajjwalkumar17, I have been having this issue o "Sending credit card numbers directly to the Stripe API is generally unsafe" with stripe. I came across you suggestion here of using `adyen`. I have properly connected adyen to hyperswich. But when I try to make a test payment in hyperswitch, I get the error message "Not allowed" as shown below;
![image](https://github.com/user-attachments/assets/74f1ae33-22b3-4bad-9748-d37a9bb6f998)

I have done research on this error and see it is caused because of passing card info in API as seen on [here](https://help.adyen.com/knowledge/ecommerce-integrations/error-codes/why-do-i-receive-a-403-or-010-not-allowed-error).

Please help and let me know on how I can make a test payment using adyen in hyperswitch hosted version. Thanks!

**Comment by Narayanbhat166** [comment]
You can enable passing raw card information on stripe by following [this reference](https://support.stripe.com/questions/enabling-access-to-raw-card-data-apis). For test mode, just send the email to stripe support. no document is required to be submitted to get this enabled, as stated in the reference
